src/cli/self_update/windows.rs

```diff
@@ -41,10 +48,102 @@ pub(crate) fn do_msvc_check(opts: &InstallOpts<'_>) -> bool {
+        let is_x86 = host_triple.contains("i686") || host_triple.contains("x86_64");
+        if is_x86 && !has_any_vs {
+            Some(VsInstallPlan::Automatic)
+        } else {
+            Some(VsInstallPlan::Manual)
+        }
+    } else {
+        None
+    }
+}
+
+#[derive(Debug)]
+struct VsInstallError(i32);
+impl std::error::Error for VsInstallError {}
+impl fmt::Display for VsInstallError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // See https://docs.microsoft.com/en-us/visualstudio/install/use-command-line-parameters-to-install-visual-studio?view=vs-2022#error-codes
+        let message = match self.0 {
+            740 => "elevation required",
+            1001 => "Visual Studio installer process is running",
+            1003 => "Visual Studio is in use",
+            1602 => "operation was canceled",
+            1618 => "another installation running",
+            1641 => "operation completed successfully, and reboot was initiated",
+            3010 => "operation completed successfully, but install requires reboot before it can be used",
+            5003 => "bootstrapper failed to download installer",
+            5004 => "operation was canceled",
+            5005 => "bootstrapper command-line parse error",
+            5007 => "operation was blocked - the computer does not meet the requirements",
+            8001 => "arm machine check failure",
+            8002 => "background download precheck failure",
+            8003 => "out of support selectable failure",
+            8004 => "target directory failure",
+            8005 => "verifying source payloads failure",
+            8006 => "Visual Studio processes running",
+            -1073720687 => "connectivity failure",
+            -1073741510 => "Microsoft Visual Studio Installer was terminated",
+            _ => "error installing Visual Studio"
+        };
+        write!(f, "{} (exit code {})", message, self.0)
     }
+}
 
-    true
+pub(crate) fn try_install_msvc() -> Result<()> {
+    // download the installer
+    let visual_studio_url = utils::parse_url("https://aka.ms/vs/17/release/vs_community.exe")?;
+
+    let tempdir = tempfile::Builder::new()
+        .prefix("rustup-visualstudio")
+        .tempdir()
+        .context("error creating temp directory")?;
+
+    let visual_studio = tempdir.path().join("vs_setup.exe");
+    utils::download_file(&visual_studio_url, &visual_studio, None, &|_| ())?;
+
+    // Run the installer. Arguments are documented at:
+    // https://docs.microsoft.com/en-us/visualstudio/install/use-command-line-parameters-to-install-visual-studio
+    let mut cmd = Command::new(visual_studio);
+    cmd.arg("--wait")
+        // Display an interactive GUI focused on installing just the selected components.
+        .arg("--focusedUi")
+        // Add the linker and C runtime libraries.
+        .args(["--add", "Microsoft.VisualStudio.Component.VC.Tools.x86.x64"]);
```

Comment:
Is this component the correct name regardless of if we're i686 or x86_64 ?

Reply:
Yes. The name means x86/x64 (i.e. "and") rather than meaning `x86_64`, if that makes sense. So basically it means both 32 and 64 bit.

Reply:
Aha OK thanks you for the clarification.